presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PlanNodeSearcher.java

```diff
@@ -31,12 +31,15 @@
 
 public class PlanNodeSearcher
 {
-    @Deprecated
     public static PlanNodeSearcher searchFrom(PlanNode node)
     {
         return searchFrom(node, noLookup());
     }
 
+    /**
+     * Use it in optimizer {@link com.facebook.presto.sql.planner.iterative.Rule} if you truely do not have a better option
+     */
+    @Deprecated
```

Comment:
we don't, so it is acceptable to use it today. But I would not hide the issue that it is not the ultimate solution and we need to think about something else in the future.